external/fv3fit/tests/training/test_autoencoder.py

```diff
@@ -0,0 +1,174 @@
+import numpy as np
+import xarray as xr
+from typing import Sequence
+from fv3fit.pytorch.cyclegan import AutoencoderHyperparameters, train_autoencoder
+from fv3fit.pytorch.cyclegan.train import TrainingConfig
+from fv3fit.tfdataset import iterable_to_tfdataset
+import collections
+import os
+import fv3fit.pytorch
+import fv3fit
+
+
+def get_tfdataset(nsamples, nbatch, ntime, nx, ny, nz):
+    ntile = 6
+
+    grid_x = np.arange(0, nx, dtype=np.float32)
+    grid_y = np.arange(0, ny, dtype=np.float32)
+    grid_x, grid_y = np.broadcast_arrays(grid_x[:, None], grid_y[None, :])
+    grid_x = grid_x[None, None, None, :, :, None]
+    grid_y = grid_y[None, None, None, :, :, None]
+
+    def sample_iterator():
+        # creates a timeseries where each time is the negation of time before it
+        for _ in range(nsamples):
+            ax = np.random.uniform(0.1, 1.5, size=(nbatch, 1, ntile, nz))[
+                :, :, :, None, None, :
+            ]
+            bx = np.random.uniform(8, 16, size=(nbatch, 1, ntile, nz))[
```

Comment:
Seems like the period should depend on the grid size instead of being hardcoded?

Reply:
Sort of, what happened was I chose periods that are only as large as they need to be in order to be "resolved enough", and then I chose grid sizes to suit those periods. If the grid size is too small to hold a whole wave, that can still be fine, possibly. But I really wouldn't want someone to put `nx=4` and then have the code say the period is `nx/2` (which would give constant data).